src/components/Asset/AssetActions/Download.tsx

```diff
@@ -236,7 +236,7 @@ export default function Download({
       disabled={baseTokenBalance === 0 || !baseTokenBalance}
       hasPreviousOrder={false}
       hasDatatoken={false}
-      dtSymbol={asset?.accessDetails?.baseToken.symbol}
+      dtSymbol={asset.accessDetails.datatoken.symbol}
```

Comment:
and now a missing datatoken symbol will break the component. The solution for all those `Cannot read properties of undefined` errors is to make correct use of optional chaining operator so in this case `asset?.accessDetails?.datatoken?.symbol` which is the same as `asset && asset.accessDetails && asset.accessDetails.datatoken && asset.accessDetails.datatoken.symbol`

This is required cause both `asset` and then `asset.accessDetails` take some time to be fetched and constructed

Reply:
but it will never reach that point if it doesn't have the datatoken symbol. On line 271, I'm checking also that the symbol is valued, otherwise it doesn't render this component (<CollectTokensButton/>). Since this parameter (dtSymbol) is required for the correct functioning of the component, I though to check if the symbol is valued before rendering it. This will prevent the component to break as it'll never render without symbol. Maybe is not the right approach?